File: rentacar-frontend/agentFront/src/app/model/Ad.ts
import { Car } from '../model/Car';
import { PriceList } from './PriceList';

export class Ad {
    id : number;
    car: Car = new Car();
    city: String = "";
    collisionDamageWaiver : boolean;
    userID: number;
    startDate: string = "";
    endDate: string = "";
    comments: Comment[] = [];
    rating: number;
    priceListID: number;
    discount: number;
    priceList: PriceList;
}

File: rentacar-backend/adService/src/main/java/com/adService/model/Ad.java
package com.adService.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;

import javax.persistence.*;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@Builder
@ToString
@AllArgsConstructor
@RequiredArgsConstructor
public class Ad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonBackReference
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Car car;

    @Column
    private String city;

    @Column
    private boolean collisionDamageWaiver;

    @Column
    private Long userID;

    @Column
    private double rating;

    @OneToMany(mappedBy = "ad", cascade = CascadeType.REFRESH)
    private Set<Comment> comments = new HashSet<>();

    @JsonBackReference
    @ManyToOne
    private PriceList priceList;

    @Column
    private Date startDate;

    @Column
    private Date endDate;

    @Column(nullable = true)
    private double discount;
}
